WooCommerce/Classes/ViewRelated/AppCoordinator.swift

```diff
@@ -90,6 +101,25 @@ private extension AppCoordinator {
             }
         }
     }
+
+    /// Displays the role error page as the root view.
+    ///
+    func displayRoleErrorUI(for siteID: Int64, errorInfo: EligibilityErrorInfo) {
+        let errorViewModel = RoleErrorViewModel(siteID: siteID, title: errorInfo.name, subtitle: errorInfo.humanizedRoles)
+        let errorViewController = RoleErrorViewController(viewModel: errorViewModel)
+
+        // when the retry is successful, resume the intention to display the main tab bar.
+        errorViewModel.onSuccess = {
+            self.displayLoggedInUI()
+        }
+
+        errorViewModel.onDeauthenticationRequest = {
+            self.stores.deauthenticate()
+        }
+
+        // this needs to be wrapped within a navigation controller to properly show the right bar button for Help.
+        setWindowRootViewControllerAndAnimateIfNeeded(UINavigationController(rootViewController: errorViewController))
```

Comment:
⚠️  I think we should use `WooNavigationController` to be safe with any future global customization.

Reply:
Ah, right! About this, I tried using `WooNavigationController` but it was [overriding the status bar style](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/System/WooNavigationController.swift#L25-L30) to `.lightContent` regardless of the current color scheme. 

I'll follow your suggestion on using `WooNavigationController` for now, and will probably raise this as a separate issue later. Thanks! 👍